mephisto/abstractions/providers/mturk/mturk_utils.py

```diff
@@ -61,21 +61,55 @@ def check_aws_credentials(profile_name: str) -> bool:
 def setup_aws_credentials(
     profile_name: str, register_args: Optional[DictConfig] = None
 ) -> bool:
+    if not os.path.exists(os.path.expanduser("~/.aws/")):
+        os.makedirs(os.path.expanduser("~/.aws/"))
+    aws_credentials_file_path = "~/.aws/credentials"
+    expanded_aws_file_path = os.path.expanduser(aws_credentials_file_path)
     try:
         # Check existing credentials
         boto3.Session(profile_name=profile_name)
         if register_args is not None:
             # Eventually we could manually re-parse the file and see
             # if the credentials line up or not, then fix ourselves
-            print(
-                f"WARNING credentials provided, but there's already a "
-                f"profile for {profile_name}. If these don't line up, you'll "
-                f"need to manually navigate to your ~/.aws/credentials file "
-                f"and remove the entry for this profile name, then run again.\n"
-                f"As this profile is currently loading, we consider it "
-                f"successfully registered anyways."
-            )
+            aws_credentials_file_string = ""
+            with open(expanded_aws_file_path, "r") as aws_credentials_file:
+                aws_credentials_file_string = aws_credentials_file.read()
+            # accessing the aws_credentials_file
+            aws_credentials = aws_credentials_file_string.split("\n")
+            # iterating to get the profile
+            profileFound = False
+            for credentialIndex in range(0, len(aws_credentials)):
+                if profileFound:
+                    aws_access_key_id = aws_credentials[credentialIndex]
+                    aws_secret_access_key = aws_credentials[credentialIndex+1]
+                    break
+                if str(aws_credentials[credentialIndex]).startswith("[{}]".format(profile_name)):
+                    profileFound = True
```

Comment:
Nit - this body could definitely be simplified:
```
for credentialIndex in range(0, len(aws_credentials)):
    if str(aws_credentials[credentialIndex]).startswith("[{}]".format(profile_name)):
        aws_access_key_id = aws_credentials[credentialIndex+1]
        aws_secret_access_key = aws_credentials[credentialIndex+2]
    break  # credentialIndex now has the index into credentials we're using.
```

Reply:
Note however that the contents of the lines in `aws_credentials[credentialIndex+1]` and `aws_credentials[credentialIndex+2]` are _not_ just the credentials, they are as follows:
```
aws_credentials_file.write(
    "aws_access_key_id={}\n".format(register_args.access_key_id)
)
```
Your code should anticipate this and look for the full string to compare to. (either by stripping the content before the first `=` or by comparing to strings containing this as well).